src/main/java/core/mate/academy/service/ExcavatorProducer.java

```diff
@@ -0,0 +1,20 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ExcavatorProducer implements MachineProducer<Machine> {
```

Comment:
Parameterize `MachineProducer` with `Excavator` instead of `Machine` to prevent implementations that do not use `Machine` or its subtype. The current implementation allows for any type to be used, which violates the checklist requirement.